### PR TITLE
[solana-receiver] Post price update

### PR DESCRIPTION
**Pull request by guibescos**
This achieves parsePriceFeeds parity by posting a PriceUpdateMessage on Solana.
The message gets posted to an ephemeral account whose rent write_authority will be able to claim.
We also support checking less than 13 signatures by posting the number of signatures that have been verified.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -108,6 +108,6 @@ repos:
       - id: cargo-clippy-pyth-solana-receiver
         name: Cargo clippy for solana target chain contract
         language: "rust"
-        entry: cargo +nightly-2023-03-01 clippy --manifest-path ./target_chains/solana/Cargo.toml --tests --fix --allow-dirty --allow-staged --features devnet -- -D warnings
```
Comment (guibescos) [inline .pre-commit-config.yaml:111]: The feature flags got killed in this PR

File: pythnet/pythnet_sdk/src/messages.rs [hunk 2]
```diff
@@ -68,7 +74,9 @@ impl Arbitrary for Message {
 pub type FeedId = [u8; 32];
 
 #[repr(C)]
-#[derive(Debug, Copy, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(
+    Debug, Copy, Clone, PartialEq, Serialize, Deserialize, BorshDeserialize, BorshSerialize,
```
Comment (guibescos) [inline pythnet/pythnet_sdk/src/messages.rs:78]: Some extra borsh here for Anchor to like this struct

File: target_chains/solana/programs/pyth-solana-receiver/Cargo.toml [hunk 3]
```diff
@@ -13,23 +13,14 @@ no-entrypoint = []
...
-mainnet = ["wormhole-anchor-sdk/mainnet"]
-devnet = ["wormhole-anchor-sdk/solana-devnet"]
 test-bpf = []
 
 [dependencies]
 anchor-lang = "0.28.0"
-wormhole-core = { git = "https://github.com/guibescos/wormhole", branch = "reisen/sdk-solana"}
-wormhole-solana = { git = "https://github.com/guibescos/wormhole", branch = "reisen/sdk-solana"}
-pyth-wormhole-attester-sdk = { path = "../../../../wormhole_attester/sdk/rust" }
-pythnet-sdk = { git = "https://github.com/pyth-network/pyth-crosschain", version = "2.0.0", features = ["strum"] }
```
Comment (guibescos) [inline target_chains/solana/programs/pyth-solana-receiver/Cargo.toml:26]: Big dependency cleanup

File: target_chains/solana/programs/pyth-solana-receiver/src/lib.rs [hunk 4]
```diff
@@ -91,72 +102,79 @@ pub mod pyth_solana_receiver {
         // TODO: update pythnet_sdk to implement BorshSerialize, BorshDeserialize
         // for MerklePriceUpdate as well as Keccak160 price_updates can be passed
         // in as Vec<MerklePriceUpdate>
-        price_updates: Vec<Vec<u8>>,
+        price_update: MerklePriceUpdate,
```
Comment (jayantk) [inline target_chains/solana/programs/pyth-solana-receiver/src/lib.rs:105]: you did the TODO

File: target_chains/solana/programs/pyth-solana-receiver/src/lib.rs [hunk 5]
```diff
@@ -91,72 +102,79 @@ pub mod pyth_solana_receiver {
...
         );
+        anchor_lang::solana_program::program::invoke(
+            &transfer_instruction,
+            &[
+                ctx.accounts.payer.to_account_info(),
+                ctx.accounts.treasury.to_account_info(),
+            ],
+        )?;
+
+        let (header, body): (Header, Body<&RawMessage>) =
```
Comment (jayantk) [inline target_chains/solana/programs/pyth-solana-receiver/src/lib.rs:134]: for my own edification: wormhole guarantees that any account owned by wormhole that can be deserialized using this function is a VAA? (i.e., the header discriminates between the different types of wormhole accounts)

**Review by jayantk — APPROVED** [review]
awesome!

File: target_chains/solana/programs/pyth-solana-receiver/src/lib.rs (hunk 5 above)
Comment (guibescos) [reply target_chains/solana/programs/pyth-solana-receiver/src/lib.rs:125]: Wormhole guarantees that any account :
- Owned by wormhole
- With the right Anchor discriminator 
Is an EncodedVaa account.
Additionally if encoded_vaa.header.verified_signatures = n > 0, it means n guardian signatures have been verfied

**Pull request merged**